bundles/automation/org.eclipse.smarthome.automation.module.media/src/main/java/org/eclipse/smarthome/automation/module/media/internal/MediaActionTypeProvider.java

```diff
@@ -0,0 +1,89 @@
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.automation.Visibility;
+import org.eclipse.smarthome.automation.module.media.handler.PlayActionHandler;
+import org.eclipse.smarthome.automation.type.ActionType;
+import org.eclipse.smarthome.automation.type.Input;
+import org.eclipse.smarthome.automation.type.ModuleType;
+import org.eclipse.smarthome.automation.type.ModuleTypeProvider;
+import org.eclipse.smarthome.automation.type.Output;
+import org.eclipse.smarthome.config.core.ConfigConstants;
+import org.eclipse.smarthome.config.core.ConfigDescriptionParameter;
+import org.eclipse.smarthome.config.core.ConfigDescriptionParameter.Type;
+import org.eclipse.smarthome.config.core.ConfigDescriptionParameterBuilder;
+import org.eclipse.smarthome.config.core.ParameterOption;
+import org.eclipse.smarthome.core.audio.AudioManager;
+
+/**
+ * This class dynamically provides the Play action type.
+ * This is necessary since there is no other way to provide dynamic config param options for module types.
+ *
+ * @author Kai Kreuzer - Initial contribution and API
+ *
+ */
+public class MediaActionTypeProvider implements ModuleTypeProvider {
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public ModuleType getModuleType(String UID, Locale locale) {
+        if ("PlayAction".equals(UID)) {
+            return getPlayActionType(locale);
+        } else {
+            return null;
+        }
+    }
+
+    @Override
+    public Collection<ModuleType> getModuleTypes(Locale locale) {
+        return Collections.singleton(getPlayActionType(locale));
+    }
+
+    private ModuleType getPlayActionType(Locale locale) {
+        return new ActionType("PlayAction", getConfigDesc(), "Play Sound", "Plays a sound file", null,
+                Visibility.VISIBLE, new ArrayList<Input>(), new ArrayList<Output>());
+    }
+
+    private List<ConfigDescriptionParameter> getConfigDesc() {
+        ConfigDescriptionParameter param = ConfigDescriptionParameterBuilder
+                .create(PlayActionHandler.PARAM_SOUND, Type.TEXT).withRequired(true).withLabel("Sound")
+                .withDescription("The sound to play").withOptions(getOptions()).build();
+        return Collections.singletonList(param);
+    }
+
+    /**
+     * This method creates one option for every file that is found in the sounds directory.
+     * as a label, the file extension is removed and the string is capitalized.
+     *
+     * @return a list of parameter options representing the sound files
+     */
+    private List<ParameterOption> getOptions() {
+        List<ParameterOption> options = new ArrayList<>();
+        File soundsDir = Paths.get(ConfigConstants.getConfigFolder(), AudioManager.SOUND_DIR).toFile();
+        if (soundsDir.isDirectory()) {
+            for (String fileName : soundsDir.list()) {
+                if (fileName.contains(".")) {
+                    String soundName = StringUtils.capitalize(fileName.substring(0, fileName.indexOf(".")));
```

Comment:
Should we use "last index of" or really the first one?

Here I come up with a question:
If I have multiple audio sinks, the one supports mp3 streams, the other one wav streams.
IIRC at the moment we have to supple a compatible format.
If I have a doorbell mp3 and wav in my sounds directory, who should choose the correct one?
As the current implementation forces the user to use the correct one, I don't know if we should cut off the file name and drop the information (as a List is used, we have two doorbell entries).